### PR TITLE
Check balance before send transaction

### LINKED ISSUE #52: [FEATURE] Check balance before attempting to send ZCN
When calling ./zwallet send - the CLI always attempts to send the ZCN amount requested. 
In cases where the wallet has too little ZCN, it takes the txn ~90seconds to fail and the miner/sharder's time has been wasted.

image
image

Acceptance criteria:
it would be more efficient for the CLI to first check the balance of the wallet internally before attempting the send ZCN txn.

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**
A brief description of the changes in this PR:

- Checks for sufficient balance before sending tokens
- Transaction does not fail even if getting balance does

Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/zwalletcli/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- 0chain:
- Other: ...

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feat/optimizeSend  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | feat/CheckBalanceForFastFailOnSend  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by Kishan-Dhakan** [comment]
Before:
image

After:
image
image

**Review by stewartie4 — APPROVED** [review]
Brilliant fix for https://github.com/0chain/zwalletcli/issues/52

File: cmd/send.go [hunk 1]
```diff
@@ -10,6 +10,26 @@ import (
 	"github.com/spf13/cobra"
 )
 
+func checkBalanceBeforeSend(tokens, fee float64) {
+	wg := &sync.WaitGroup{}
+	statusBar := &ZCNStatus{wg: wg}
+	wg.Add(1)
+	err := zcncore.GetBalance(statusBar)
+	if err != nil {
+		return // continue sending txn even if getBalance fails
```
Comment (stewartie4) [inline cmd/send.go:19]: what does "fails" mean?
A balance of zero will return a 400 right now, this is a client side "Failure"
(This is not ideal and a balance of zero should be returned with a 200 instead)

Comment (Kishan-Dhakan) [reply cmd/send.go:19]: Yes fail is meant to suggest any response other than 200, the idea being that if for some reason getBalance fails, it should not also stop the user from being able to do the the transaction.

**Pull request merged**